### DISCUSSION TITLE
Starting unknown processes

### DISCUSSION THREAD

**Discussion by mstee1** [bluenviron/gortsplib#303] [General]
When you start capturing a stream from rtsp server, 8 goroutines work before calling the gortsplib.Client.Play(nil) method. After calling the gortsplib.Client.Play(nil) method, after a few seconds, third-party processes begin to appear in htop and their number reaches 14-20 within 15 minutes. What is the reason for the emergence of new processes?

**Comment by aler9** [comment]
Hello, this library is open source and you can see which routines are started and why by looking at the source code.
Furthermore, there are tools like [pprof](https://pkg.go.dev/net/http/pprof) that allows to list all routines and their entry point.
